rust/ql/src/queries/telemetry/RustAnalyzerComparison.qll

```diff
@@ -0,0 +1,134 @@
+/**
+ * INTERNAL: Do not use.
+ *
+ * Provides functionality for comparing data from `rust-analyzer` with data computed
+ * in QL.
+ */
+
+import rust
+
+private signature module ResolvableSig {
+  class Source {
+    string toString();
+
+    Location getLocation();
+  }
+
+  class Target {
+    string toString();
+
+    Location getLocation();
+  }
+}
+
+private signature module CompareSig<ResolvableSig R> {
+  predicate isResolvable(R::Source s);
+
+  R::Target resolve(R::Source s);
+}
+
+private module Compare<ResolvableSig R, CompareSig<R> RustAnalyzer, CompareSig<R> Ql> {
+  private import R
+
+  predicate same(Source s, Target t) {
+    t = RustAnalyzer::resolve(s) and
+    t = Ql::resolve(s)
+  }
+
+  predicate sameCount(int c) { c = count(Source s | same(s, _)) }
+
+  predicate diff(Source s, Target t1, Target t2) {
+    t1 = RustAnalyzer::resolve(s) and
+    t2 = Ql::resolve(s) and
+    t1 != t2
+  }
+
+  predicate diffCount(int c) { c = count(Source s | not same(s, _) and diff(s, _, _)) }
```

Comment:
Why do we exclude those that are in `same(s, _)` here? If there's overlap I guess it's because one of the resolvers produce several results for a `Source`.